reconcile/terraform_repo.py

```diff
@@ -0,0 +1,244 @@
+import logging
+from collections.abc import (
+    Callable,
+    Mapping,
+    MutableMapping,
+)
+from typing import (
+    Any,
+    Optional,
+)
+
+import yaml
+from pydantic import BaseModel
+
+from reconcile import queries
+from reconcile.gql_definitions.terraform_repo.terraform_repo import (
+    TerraformRepoV1,
+    query,
+)
+from reconcile.utils import gql
+from reconcile.utils.defer import defer
+from reconcile.utils.exceptions import ParameterError
+from reconcile.utils.gitlab_api import GitLabApi
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.state import (
+    State,
+    init_state,
+)
+
+QONTRACT_INTEGRATION = "terraform-repo"
+QONTRACT_INTEGRATION_VERSION = make_semver(1, 1, 0)
+
+
+class ActionPlan(BaseModel):
+    dry_run: bool
+    repos: list[TerraformRepoV1]
+
+
+def get_repos(query_func: Callable) -> list[TerraformRepoV1]:
+    """Return all terraform repos defined in app-interface"""
+    query_results = query(query_func=query_func).repos
+    if query_results:
+        return query_results
+    return []
+
+
+def get_existing_state(state: State) -> list[TerraformRepoV1]:
+    """Get the existing state of terraform repos from S3"""
+    repo_list: list[TerraformRepoV1] = []
+    keys = state.ls()
+    for key in keys:
+        if value := state.get(key.lstrip("/"), None):
+            repo = TerraformRepoV1.parse_raw(value)
+
+            if repo is not None:
```

Comment:
I think `repo` can't be `None` here. This check isn't needed

Reply:
You're right. The `parse_raw` function does not return `None`, it returns a `ValidationError` exception instead